main.cpp

```diff
@@ -121,357 +163,390 @@ struct CarWash
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+
+bool Envelope::receiveGate(bool cycleModeOn)
+{
+    if (cycleModeOn == false)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
 
+void Envelope::cycleMode(bool switchOn)
+{
+    if (switchOn == true)
+    {
+        switchOn = false;
+    }
+    else
+    {
+        switchOn = true;
+    }
+}
 
-    //1)  receive cv
-    void receiveCV(int cvInputs);
-    //2)  output audio/cv
-    void output();
-    //3)  mute output
-    void mute(bool muteSwitch);
+struct DualVCA
+{
+    int audioInputs = 2;
+    int cvInputs = 2;
+    bool cvInputUsed = false;
+    bool boostSwitch = true;
+    int audioOutputs = 2;
+    float cvInputOne = 12.02f;
+    float cvInputTwo = 0.0f;
+
+    void cascadeCV();
+    void boost();
+    void mute(bool muteOn);
 };
+void DualVCA::cascadeCV()
+{
+    for (int i = 0; i <= cvInputs; ++i)
+    {
+        if (cvInputUsed == false)
+        {
+            cvInputTwo = cvInputOne;
+        }
+    }
+}
+
+
+void DualVCA::boost()
+{
+    if (boostSwitch == false)
+    {
+        boostSwitch = true;
+    }
+    else
+    {
+        boostSwitch = false;
+    }
+}
+
+void DualVCA::mute(bool muteOn)
+{
+    if (muteOn == true)
+    {
+        muteOn = false;
+    }
+    else
+    {
+        muteOn = true;
+    }
+}
+
 
-/*
-Thing 9) CV Sequencer
-5 properties:
-    1)  number of channels
-    2)  channel output level
-    3)  number of steps
-    4)  clock mode(slave or master)
-    5)  scale select (int)
-3 things it can do:
-    1)  output cv
-    2)  receive clock
-    3)  arpeggiate steps
- */
 struct CVSequencer
 {
-    //1)  number of channels
     int cvChannels = 8;
-    //2)  channel output level
     float outputLevel = 2.3f;
-    //3)  number of steps
     int steps = 16;
-    //4)  clock mode(slave or master)
-    bool clockMode = false;
-    //5)  scale select
+    int trigPerStep = 1;
     int scale = 2;
 
-    //1)  output cv
-    bool output(bool powerOn);
-    //2)  receive clock
-    bool receiveClock(bool clockMode);
-    //3)  arpeggiate steps
-    bool arp(bool stepsOn, std::vector<int> stepOrder[]);
+    void changeNumSteps(int newNumSteps);
+    void divideClock(int division);
+    std::vector<int> arp(int numStepsPressed);
 };
 
-/*
-Thing 10) Modular Synth
-5 properties:
-    1)  oscillators
-    2)  filters
-    3)  envelopes
-    4)  VCAs    
-    5)  CV Sequencer
-3 things it can do:
-    1)  changes pitch
-    2)  cross modulates
-    3)  modulate decay time of AD envelope
- */
+void CVSequencer::changeNumSteps(int newNumSteps)
+{
+    steps = newNumSteps;
+}
+
+void CVSequencer::divideClock(int division)
+{
+    trigPerStep *= division;
+}
+
+std::vector<int> CVSequencer::arp(int numStepsPressed)
+{
+    std::vector<int> arp; 
+    for (int i = 0; i <= numStepsPressed; ++i)
+    {
+        int stepNum;
+        std::cin >> stepNum;
+        arp.push_back(stepNum);
+    }
```

Comment:
Do not use `cin` in any of these projects.  
- [ ] revise.

Reply:
This is one of the functions i may have gotten a little overambitious on. what would you suggest i write for this? in terms of telling the function to allow the user to input steps one at a time?